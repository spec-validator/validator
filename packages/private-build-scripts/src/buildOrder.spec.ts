import {
  getWorkspaceInfo,
  getGraph,
  getPackageNamesInBuildOrder,
  getProjectsPathsInBuildOrder,
} from './buildOrder'

const WORKSPACE_INFO = {
  'pkg1': {
    location: './pkg1',
    workspaceDependencies: [],
  },
  'pkg2': {
    location: './pkg2',
    workspaceDependencies: ['pkg1'],
  },
  'pkg3': {
    location: './pkg3',
    workspaceDependencies: ['pkg2'],
  },
  'pkg4': {
    location: './pkg4',
    workspaceDependencies: ['pkg3'],
  },
}

jest.mock('./getOutput', () => ({
  default: () => JSON.stringify(WORKSPACE_INFO),
  __esModule: true,
}))

test('getWorkspaceInfo', () => {
  expect(getWorkspaceInfo()).toEqual(WORKSPACE_INFO)
})

test(
  'getGraph',
  () => expect(getGraph()).toMatchSnapshot()
)

test(
  'getPackageNamesInBuildOrder',
  () => expect(getPackageNamesInBuildOrder()).toMatchSnapshot()
)

test(
  'getProjectsPathsInBuildOrder',
  () => expect(getProjectsPathsInBuildOrder()).toMatchSnapshot()
)
