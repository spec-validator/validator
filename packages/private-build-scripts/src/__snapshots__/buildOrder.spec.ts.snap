// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGraph 1`] = `
Object {
  "pkg1": Array [],
  "pkg2": Array [
    "pkg1",
  ],
  "pkg3": Array [
    "pkg2",
  ],
  "pkg4": Array [
    "pkg3",
  ],
}
`;

exports[`getPackageNamesInBuildOrder 1`] = `
Array [
  "pkg4",
  "pkg3",
  "pkg2",
  "pkg1",
]
`;

exports[`getProjectsPathsInBuildOrder 1`] = `
Array [
  "./pkg4",
  "./pkg3",
  "./pkg2",
  "./pkg1",
]
`;
